libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -322,14 +322,26 @@ internal class MapboxTripSession(
             val routeCoordinates = tripStatus.route?.routeOptions()?.coordinatesList()
             return if (routeCoordinates != null) {
                 val waypointsCount = routeCoordinates.size
-                // TODO: get rid of the workaround https://github.com/mapbox/mapbox-navigation-android/issues/5275
-                val nextWaypointIndex = max(
-                    MIN_NEXT_WAYPOINT_INDEX,
+                val nextWaypointIndex = normalizeNextWaypointIndex(
                     tripStatus.navigationStatus.nextWaypointIndex
                 )
                 return waypointsCount - nextWaypointIndex
             } else 0
         }
+
+        /**
+         * On the Android side, we always start navigation from the current position.
+         * So we expect that the next waypoint index will not be less than 1.
+         * But the native part considers the origin as a usual waypoint.
```

Comment:
@averkhaturau, did I understand you correctly? Is that really what's going on in native part?

Reply:
He confirmed that in the slack message 🙂